ext/standard/basic_functions.stub.php

```diff
@@ -768,6 +768,140 @@ function scandir(string $directory, int $sorting_order = 0, $context = UNKNOWN):
 function glob(string $pattern, int $flags = 0): array|false {}
 #endif
 
+/* exec.c */
+
+function exec(string $command, &$output = null, &$result_code = null): string|false {}
+
+function system(string $command, &$result_code = null): string|false {}
+
+function passthru(string $command, &$result_code = null): string|null|false {}
+
+function escapeshellcmd(string $command): string {}
+
+function escapeshellarg(string $arg): string {}
+
+function shell_exec(string $command): string|false|null {}
```

Comment:
This is ugly -- I'm wondering if we can return "" instead of null. But I'm afraid this might actually break some code...

Reply:
https://www.php.net/manual/en/function.shell-exec.php explicitly mentions `null` as an error condition (but not `false`!). So maybe could we return `null` instead of `false`?